apps/src/templates/sectionsRefresh/SectionsSetUpContainer.jsx

```diff
@@ -186,6 +134,65 @@ export default function SectionsSetUpContainer({sectionToBeEdited}) {
     }
   };
 
+  const saveSection = section => {
+    // Determine data sources and save method based on new vs edit section
+    const dataUrl = isNewSection
+      ? SECTIONS_API
+      : `${SECTIONS_API}/${section.id}`;
+    const method = isNewSection ? 'POST' : 'PATCH';
+    const loginType = isNewSection
+      ? queryParams('loginType')
+      : section.loginType;
+    const participantType = isNewSection
+      ? queryParams('participantType')
+      : section.participantType;
+
+    const form = document.querySelector(`#${FORM_ID}`);
+    // If we find a missing field in the form, report which one and reset save status
+    if (!form.checkValidity()) {
+      form.reportValidity();
+      setIsSaveInProgress(false);
+      return;
+    }
+
+    const csrfToken = document.querySelector('meta[name="csrf-token"]')
+      .attributes['content'].value;
+
+    const section_data = {
+      login_type: loginType,
+      participant_type: participantType,
+      course_offering_id: section.course?.courseOfferingId,
+      course_version_id: section.course?.versionId,
+      unit_id: section.course?.unitId,
+      restrict_section: section.restrictSection,
+      lesson_extras: section.lessonExtras,
+      pairing_allowed: section.pairingAllowed,
+      tts_autoplay_enabled: section.ttsAutoplayEnabled,
+      sharing_disabled: section.sharingDisabled,
+      grades: section.grade,
+      ...section,
+    };
+
+    fetch(dataUrl, {
+      method: method,
+      headers: {
+        'Content-Type': 'application/json',
+        'X-CSRF-Token': csrfToken,
+      },
+      body: JSON.stringify(section_data),
+    })
+      .then(response => response.json())
+      .then(data => {
+        recordSectionSetupEvent(section);
+        // Redirect to the sections list.
+        window.location.href = window.location.origin + '/home';
+      })
+      .catch(err => {
+        setIsSaveInProgress(false);
+        console.error(err);
```

Comment:
I checked how we do error messaging in other files, and this is a standard pattern. Happy to hear input on better/more helpful messaging!

Reply:
This seems fine for now, though it wouldn't give any feedback to the user. We should probably (in a followup) figure out a way to show them the error if saving fails